core/wiki/macros/tree.tid

```diff
@@ -1,53 +1,62 @@
 title: $:/core/macros/tree
 tags: $:/tags/Macro
 
-\define leaf-node(prefix)
+\define link(full-title, chunk)
+<$link to=<<__full-title__>>>
+<$text text=<<__chunk__>>/>
+</$link>
+\end
+
+\define leaf-node(prefix, chunk)
 <li>
-<$list filter="""[[$prefix$$(chunk)$]is[shadow]] [[$prefix$$(chunk)$]is[tiddler]] +[removeprefix[$prefix$]] +[limit[1]]""" 
-emptyMessage="""<$text text="$prefix$$(chunk)$"/>""">
-<span>{{$:/core/images/file}}</span> <$link to="""$prefix$$(chunk)$""">
-<$view field="title"/>
-</$link> 
+<$list filter="""[[$prefix$$chunk$]is[shadow]] [[$prefix$$chunk$]is[tiddler]]""" variable="full-title">
+<$list filter="""[<full-title>] +[removeprefix<__prefix__>]""" variable="chunk">
+<span>{{$:/core/images/file}}</span>
+<$macrocall $name="link" full-title=<<full-title>> chunk=<<chunk>>/>
+</$list>
 </$list>
 </li>
 \end
 
-\define branch-node(prefix)
-<$reveal type="nomatch" text="" default="""$(chunk)$""">
-<li>      
-<$list filter="""[all[shadows+tiddlers]prefix[$prefix$$(chunk)$]] -[[$prefix$$(chunk)$]] +[limit[1]]""">
-<$reveal type="nomatch" state="""$:/state/tree/$prefix$$(chunk)$""" text="show">
-<$button set="""$:/state/tree/$prefix$$(chunk)$""" setTo="show" class="tc-btn-invisible">{{$:/core/images/folder}} <$text text="""$(chunk)$"""/></$button>
+\define branch-node(prefix, chunk)
+<li>
+<$reveal type="nomatch" state="""$:/state/tree/$prefix$$chunk$""" text="show">
```

Comment:
here we could use `state={{{ [[$:/state/tree/]addsuffix<__prefix__>addsuffix<__chunk__>] }}}` or we define the state with a `vars` widget here above like `<$vars stateName={{{ [[$:/state/tree/]addsuffix<__prefix__>addsuffix<__chunk__>] }}}>` so that we can reuse it below?